cpp/include/cudf/strings/regex/config.hpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/strings/regex/flags.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+
+namespace cudf {
+namespace strings {
+
+/**
+ * @addtogroup strings_regex
+ * @{
+ */
+
+/**
+ * @brief Compute working memory size for evaluating a regex pattern
+ * for a given strings column.
+ *
+ * This computes only the state data memory size required to process
+ * a regex pattern over the given strings column.
+ * Specific functions that use regex may require additional working memory
+ * unrelated to the regex processing.
+ *
+ * @param input Strings instance
+ * @param pattern Regex pattern to be used
+ * @param flags Regex flags for interpreting special characters in the pattern
+ * @return Size of the state memory in bytes required for processing `pattern` on `strings`
+ *         and the number of concurrent that memory will support
+ */
+std::pair<std::size_t, size_type> compute_regex_state_memory(
+  strings_column_view const& input,
+  std::string const& pattern,
```

Comment:
```suggestion
  std::string_view pattern,
```

Reply:
I want to make this change in a follow-on PR. I noticed recently all our regex APIs accept a `std::string const` parameter for the pattern: https://docs.rapids.ai/api/libcudf/stable/group__strings__contains.html
This is likely because it was coded before `std::string_view` existed (C++14 I think).

Turns out changing this is a little more involved because deeper classes require `std::string const`. This means I'd have to convert an input `std::string_view` to `std::string` internally to get it to compile. 
So I want to fix this and change all the APIs at once and did not want to include such a change in this PR.

Reply:
#10810